ampersand-state.js

```diff
@@ -386,8 +386,15 @@ _.extend(Base.prototype, BBEvents, {
                 options = options || {};
 
                 var newVal = def.fn.call(self);
+                var newValToCompare = newVal;
```

Comment:
Suggestion: Rather than setting these vars up here, and possibly re-defining later. It's a little bit clearer (and a little faster) to just declare these vars here, and then define them in an `else`.